File: src/main/java/cloud/common/Generator.java
package cloud.common;

import instance.common.Request;
import cloud.requestengine.RequestDone;
import cloud.requestengine.RequestGeneratorInit;
import se.sics.kompics.PortType;

/**
 * 
 * @author Amir Moulavi
 * @date 2011-03-30
 *
 */

public class Generator extends PortType {
	{
		negative(RequestGeneratorInit.class);
		negative(RequestDone.class);
		negative(SendRawData.class);
		positive(SendRawData.class);
		positive(Request.class);
	}
}

File: src/main/java/cloud/common/Generator.scala
package cloud.common

import se.sics.kompics.PortType
import cloud.requestengine.{DownloadStarted, RequestGeneratorInit}
import cloud.elb.BlocksActivated
import instance.common.Request

/**
 * @author Amir Moulavi
 * @date 2011-07-09
 *
 */
class Generator extends PortType {

  negative(classOf[RequestGeneratorInit])
  negative(classOf[DownloadStarted])
  negative(classOf[DownloadRejected])
  negative(classOf[SendRawData])
  negative(classOf[BlocksActivated])
  positive(classOf[SendRawData])
  positive(classOf[Request])

}